x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -394,19 +395,31 @@ export class Plugin implements ISecuritySolutionPlugin {
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     const exceptionListClient = this.lists!.getExceptionListClient(savedObjectsClient, 'kibana');
 
+    const { authz, agentService, packageService, packagePolicyService, agentPolicyService } =
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      plugins.fleet!;
+
     this.endpointAppContextService.start({
-      agentService: plugins.fleet?.agentService,
-      packageService: plugins.fleet?.packageService,
-      packagePolicyService: plugins.fleet?.packagePolicyService,
-      agentPolicyService: plugins.fleet?.agentPolicyService,
+      fleetAuthzService: authz,
+      agentService,
+      packageService,
+      packagePolicyService,
+      agentPolicyService,
       endpointMetadataService: new EndpointMetadataService(
         core.savedObjects,
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        plugins.fleet?.agentPolicyService!,
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        plugins.fleet?.packagePolicyService!,
```

Comment:
👍 on removing the `disable-next-line`

Reply:
@FrankHassanabad ,
I been meaning to bring this up with the security solution platform team. Should these plugins still be "optional"? I'm thinking that we reached a point in the maturity of the security solution plugin ++ endpoint that it does not make sense to have `fleet` be optional (or `lists` or `security` for that matter).